### ISSUE TITLE
Suggestion: Add a login page

### ISSUE THREAD

**Issue by tomyslo1** [gameyfin/gameyfin#26]
Hi, I think that a login page or an option to enable it would be great so not everyone can take up your bandwidth by downloading your library.

**Comment by grimsi** [comment]
I did not implement a login page since I intended Gameyfin only to be hosted on the local network and not publicly accessible. But if there are users who want to see this feature I could implement it.

**Comment by Zorograc** [comment]
Yeah, a login page would be awesome!

**Comment by groque** [comment]
Would be great to have a login page

**Comment by madeofpendletonwool** [comment]
+1 to the login page. I would love to give friends login access to download games from my library.

**Comment by DebugDax** [comment]
+1 Here as well, would love to share my page with friends via reverse proxy but not have the whole internet have access to my library.

**Comment by Pfuenzle** [comment]
I agree that this would be a great feature, but could you please add a function to enable login via a HTTP header?
I handle auth currently using key leak, so users have to authenticate there before being able to access the site.
Key cloak injects a `X-User: <username>` header to the request. This could be ready by Gameyfin and automatically log into the user and create an account if none is existing.
This may sound insecure, but nginx can strip those headers if manually set by the user, so it's secure. Also many other apps support this, like organizr and calibre-web. 

If you don't want to add this it would be great if the account system is optional to view and download games, like it is currently

**Comment by grimsi** [comment]
My plan is to add a config option like `gameyfin.public-access.enabled: true|false` to enable or disable anonymous read-only access (the way it is now).
So you can share your game library public, but if you want to mark games as played, bookmark games or upload save files, you have to create an account.

**Comment by puregreen59** [comment]
+1 for login page! i have remote family, dont want just anyone to be able to see and grab my games :)

**Comment by BinaryGenius** [comment]
I think it we should just react to one message instead of doing all the +1 stuff it cleans up the issue nicely. Just a thought, great repo by the way.

**Comment by Lixar91** [comment]
That would be freaking awesome. It could be also a good idea to create an admin site for the admin to manage all the accounts.

**Comment by alexfornuto** [comment]
I think the last comment speaks to the undiscussed part of this feature: multiple users and user management. I'd also like to use this tool to over the internet (proxied with TLS and all that), and would like to have multiple users as part of that setup.

**Comment by grimsi** [comment]
Just a quick update on your issue. Login page and SSO support is coming in v2.
See here: https://github.com/gameyfin/gameyfin/issues/234#issuecomment-2353193364
